docs/auth.md

```diff
@@ -478,12 +492,30 @@ Kubernetes `Secrets`.
          name: demo-pipeline
      ```
 
-1. Execute the build:
+4. Execute the build:
 
    ```shell
    kubectl apply --filename secret.yaml --filename serviceaccount.yaml --filename taskrun.yaml
    ```
 
+### In-Pod Docker Authentication
+In addition to image pulling authentication, Tekton also sets up Docker 
+authentication within the Pod's container environment. This authentication setup 
+allows the Steps within the `Pod` to interact with Docker registries during 
+execution, enabling operations like pushing or pulling images, or performing 
+other OCI image manipulations.
+
+After the Pod is created and the container images are pulled, Tekton generates 
+a `~/.docker/config.json` file within the container. This file contains the 
+necessary Docker authentication credentials, allowing the Steps to authenticate 
+with Docker registries during execution.
+
+The Docker authentication credentials used for in-Pod authentication are derived
+from the same Secrets specified as `imagePullSecrets`. Tekton follows the credential 
```

Comment:
For the `in-pod` authentication (aka tekton creating `${HOME}/.docker/config.json`, it does use `secrets` and not `imagePullSecrets`. This is the core of the confusion to be honest 🙃, `imagePullSecrets` is used by Kubernetes to pull images to be run, `secrets` is used by Tekton to create the "docker" authentication configuration for something in-the-pod to be able to access the registries. So here it should be about `secrets` and not `imagePullSecrets`